File: plugin/src/main/scala/agilesitesng/wem/model/CSElement.scala
package agilesitesng.wem.model

/**
 * Created by msciab on 16/07/15.
 */

import java.io.File
import java.util.Date

object CSElement {

  import WemModel._

  def blobFromFile(file: File): Blob = {
    val filename = file.getName
    val foldername = if (file.getParentFile != null) file.getParentFile.getName else ""
    Blob(filename, foldername, base64file(file))
  }

  def blobFromFile(filename: String): Blob = blobFromFile(new java.io.File(filename))

  def apply(_id: Long, _name: String, blob: Blob)(implicit _site: String) =
    Asset(id = s"CSElement:${_id}"
      , subtype = ""
      , name = _name
      , publist = List(_site)
      , attribute = List(
        Attribute("rootelement", Data(stringValue = Some(s"${_site}/${_name}")))
        , Attribute("elementname", Data(stringValue = Some(s"${_site}/${_name}")))
        , Attribute("resdetails1", Data(stringValue = Some(s"eid=${_id}")))
        , Attribute("url", Data(blobValue = Some(blob))
        )
      ))

}

File: plugin/src/main/java/agilesitesng/services/CSElement.java
package agilesitesng.services;

import java.util.List;

import COM.FutureTense.Interfaces.ICS;
import agilesites.api.Log;
import com.fatwire.assetapi.data.BlobObject;
import com.fatwire.assetapi.data.BlobObjectImpl;
import com.fatwire.assetapi.data.MutableAssetData;

/**
 * CSElement definition class
 */
public class CSElement extends Asset {

    final Log log = Log.getLog(CSElement.class);

    private String fileName;
    private String folder;
    private String elementCode;
    private String elementName;

    /**
     * Create a CSElement invoking the given elementClass
     */
    public CSElement(ICS ics) {
        super("CSElement", ics);
        this.fileName = ics.GetVar("filename");
        this.folder = ics.GetVar("folder");
        this.elementCode = ics.GetVar("body");
        this.elementName = ics.GetVar("name");
    }

    public List<String> getAttributes() {
        return Utils.listString("name", "description", "elementname",
                "rootelement", "url", "resdetails1", "resdetails2");
    }

    public void setData(MutableAssetData data) {

        // root element
        data.getAttributeData("rootelement").setData(elementName);
        // addAttribute(data, "rootelement", element);

        // element name
        data.getAttributeData("elementname").setData(elementName);
        // addAttribute(data, "elementname", element);

        data.getAttributeData("resdetails1").setData(
                "eid=" + data.getAssetId().getId());

        data.getAttributeData("resdetails2").setData(
                "timestamp=" + System.currentTimeMillis());

        // blob
        log.info("filename="+filename+" folder="+folder);
        byte[] bytes = elementCode.getBytes();
        BlobObject blob = new BlobObjectImpl(fileName, folder, bytes);
        data.getAttributeData("url").setData(blob);

        // data.getAttributeData("createdby").setData("agilesites");
        // data.getAttributeData("createddate").setData(new Date());
        // data.getAttributeData("Mapping").setData(new ArrayList());
        // data.getAttributeData("Mapping").setData(new AttributeMan HashMap());
    }

}
